docs/rules/match-default-export-name.md

```diff
@@ -0,0 +1,77 @@
+# import/match-default-export-name
+
+Reports use of import name what doesn't match the default export name. Does nothing for anonymous default exports.
+
+## Rule Details
+
+Given:
+```js
+// foo.js
+export default function myFunction() {}
+```
+
+...this would be valid:
+```js
+import myFunction from './foo.js';
+```
+
+...and this would be reported:
+```js
+// message: Expected import 'myFn' to match the default export 'myFunction'.
+import myFn from './foo.js';
+```
+
+Also works for post-ES2015 `export` extensions:
+
+valid:
+```js
+export myFunction from './foo.js';
+```
+
+reported:
+```js
+// message: Expected export 'myFn' to match the default export 'myFunction'.
+export myFn from './foo.js';
+```
+
+## Options
+
+This rule supports the following options:
+
+### `ignore: [array of string]`:
+
+This defines patterns to ignore files. Defaults to `["/node_modules/"]`.
+
+### `overrides: [array of objects]`:
+
+This defines custom default import names for specific modules.
+
+Properties of the objects
+
+| property  | required | type   | description   |
+|-----------|:--------:|--------|---------------|
+| module    |     x    | string | module name to match |
+| name      |     x    | string | default import name pattern |
+| transform |          | string | transform default import name pattern to given case, can be 'camelCase', 'PascalCase' or 'snake_case' |
+
+```json
+{
+  "import/match-default-export-name": ["error", {
+    "overrides": [
+      {
+        "module": "react",
+        "name": "React"
+      },
+      {
+        "module": "/styles\\.css$/",
```

Comment:
i'm not comfortable supplying regex in configs; that leads to CVEs. Let's change this to use globs, instead.